drivers/spi/spi_ll_stm32.c

```diff
@@ -451,11 +624,72 @@ static int transceive(struct device *dev,
 	return ret;
 }
 
+#ifdef CONFIG_SPI_STM32_DMA
+static int transceive_dma(struct device *dev,
+		      const struct spi_config *config,
+		      const struct spi_buf_set *tx_bufs,
+		      const struct spi_buf_set *rx_bufs,
+		      bool asynchronous, struct k_poll_signal *signal)
+{
+	const struct spi_stm32_config *cfg = DEV_CFG(dev);
+	struct spi_stm32_data *data = DEV_DATA(dev);
+	SPI_TypeDef *spi = cfg->spi;
+	int ret;
+
+	if (!tx_bufs && !rx_bufs) {
+		return 0;
+	}
+
+	if (asynchronous) {
+		return -ENOTSUP;
+	}
+
+	spi_context_lock(&data->ctx, asynchronous, signal);
+
+	data->dma_tx.transfer_complete = false;
+	data->dma_rx.transfer_complete = false;
+
+	ret = spi_stm32_configure(dev, config);
+	if (ret) {
+		return ret;
+	}
+
+	/* Set buffers info */
+	spi_context_buffers_setup(&data->ctx, tx_bufs, rx_bufs, 1);
+
+	ret = spi_dma_move_buffers(dev);
+	if (ret) {
+		return ret;
+	}
+
+	LL_SPI_Enable(spi);
+
+	do {
+	} while (spi_stm32_dma_transfer_ongoing(data));
+
+	/* This is turned off in spi_stm32_complete(). */
+	spi_context_cs_control(&data->ctx, true);
+
+	spi_context_release(&data->ctx, ret);
+
+	return ret;
+}
+#endif /* CONFIG_SPI_STM32_DMA */
+
 static int spi_stm32_transceive(struct device *dev,
 				const struct spi_config *config,
 				const struct spi_buf_set *tx_bufs,
 				const struct spi_buf_set *rx_bufs)
 {
+#ifdef CONFIG_SPI_STM32_DMA
+	struct spi_stm32_data *data = DEV_DATA(dev);
+
+	if ((data->dma_tx.dma_name != NULL)
+	 && (data->dma_rx.dma_name != NULL)) {
+		return transceive_dma(dev, config, tx_bufs, rx_bufs,
+				false, NULL);
```

Comment:
indentation, it's below last '(' + 1 space (if you configure your editor to follow linux code style it will do it for you)

Reply:
ok